docs/release-steps.md

```diff
@@ -0,0 +1,99 @@
+[base/kustomization.yaml]: ../manifests/base/kustomization.yaml
+[quay repo]: https://quay.io/repository/opendatahub/odh-dashboard?tab=tags
+[drafting a new release]: https://github.com/opendatahub-io/odh-dashboard/releases/new
+[semver]: https://semver.org/
+
+# Releases
+
+There are two types of releases in the Dashboard, and they usually happen together one after another.
+
+1. `main` (stable branch) Release - good for adoption of downstream components
+2. `incubation` (bleeding edge) Release - good for ODH & community involvement
+
+## `main` Release
+
+* Start by [drafting a new release]
+* Pick a new [release version](#version-numbers)
+  * `vX.YY.Z` based on what is included (eg. `v1.23.4`)
+  * If you do not have enough information to make the call on minor or patch version updating, you may need to pick one and wait until the release notes are generated
+* Set the release title to the same name as the tag
+* Click the `Generate release notes` button to get the full list of PRs merged into `main`
+  * Scan the notes it generated for any gaps in your Release
+  * If you were waiting on full determination of release number, make that call now
+    * Verify the tag & the title of the release; these should match
+* Add a header section at the top of the readme called `## Notable Changes`
+  * Add notes on what can be considered a high level set of changes, don't be too specific on changes and no need to catch every change - just notable ones
+  * See previous releases for examples
+* Mark the release as the `latest` and submit the release
+
+## `incubation` Release
+
+> Note: This is a two-day process until we can get the CI working for `incubation`.
+
+### Day 1
+
+> Note: The day of a `main` release is ideal, but it can use the tag if more commits occur in between. Ideally follow the exact release to maintain purity of the release.
+
+* First make sure [main is released](#main-release)
+* Merge the `main` release content into `incubation`
```

Comment:
This can be done separately... this documents the flow today and not looking to solve CI gaps. Feel free to log something and address it concurrently. There will be a need to update the docs more later and we can fix this when we have the CI in place.